sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/concatenate_data/combine_cloud_handler.hpp

```diff
@@ -0,0 +1,134 @@
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * $Id: concatenate_data.cpp 35231 2011-01-14 05:33:20Z rusu $
+ *
+ */
+
+#ifndef AUTOWARE__POINTCLOUD_PREPROCESSOR__CONCATENATE_DATA__COMBINE_CLOUD_HANDLER_HPP_
+#define AUTOWARE__POINTCLOUD_PREPROCESSOR__CONCATENATE_DATA__COMBINE_CLOUD_HANDLER_HPP_
+
+#include <deque>
+#include <memory>
+#include <mutex>
+#include <set>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+// ROS includes
+#include "autoware_point_types/types.hpp"
+
+#include <diagnostic_updater/diagnostic_updater.hpp>
+#include <point_cloud_msg_wrapper/point_cloud_msg_wrapper.hpp>
+
+#include <diagnostic_msgs/msg/diagnostic_status.hpp>
+#include <geometry_msgs/msg/twist_stamped.hpp>
+#include <geometry_msgs/msg/twist_with_covariance_stamped.hpp>
+#include <nav_msgs/msg/odometry.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <sensor_msgs/point_cloud2_iterator.hpp>
+#include <tier4_debug_msgs/msg/int32_stamped.hpp>
+#include <tier4_debug_msgs/msg/string_stamped.hpp>
+
+#include <message_filters/pass_through.h>
+#include <message_filters/subscriber.h>
+#include <message_filters/sync_policies/approximate_time.h>
+#include <message_filters/sync_policies/exact_time.h>
+#include <message_filters/synchronizer.h>
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/transform_listener.h>
+
+namespace autoware::pointcloud_preprocessor
+{
+
+class CombineCloudHandler
+{
+private:
+  rclcpp::Node * node_;
+  tf2_ros::Buffer tf_buffer_;
+  tf2_ros::TransformListener tf_listener_;
+
+  std::vector<std::string> input_topics_;
+  std::string output_frame_;
+  bool is_motion_compensated_;
+  bool keep_input_frame_in_synchronized_pointcloud_;
+
+  struct RclcppTimeHash_
+  {
+    std::size_t operator()(const rclcpp::Time & t) const
+    {
+      return std::hash<int64_t>()(t.nanoseconds());
+    }
+  };
+
+public:
+  std::deque<geometry_msgs::msg::TwistStamped::ConstSharedPtr> twist_ptr_queue_;
+
+  CombineCloudHandler(
+    rclcpp::Node * node, std::vector<std::string> input_topics, std::string output_frame,
+    bool is_motion_compensated, bool keep_input_frame_in_synchronized_pointcloud);
+  void processTwist(const geometry_msgs::msg::TwistWithCovarianceStamped::ConstSharedPtr & input);
+  void processOdometry(const nav_msgs::msg::Odometry::ConstSharedPtr & input);
+
+  std::tuple<
+    sensor_msgs::msg::PointCloud2::SharedPtr,
+    std::unordered_map<std::string, sensor_msgs::msg::PointCloud2::SharedPtr>,
+    std::unordered_map<std::string, double>>
+  combinePointClouds(std::unordered_map<std::string, sensor_msgs::msg::PointCloud2::SharedPtr> &
```

Comment:
This should probably be its own type for clarity

Reply:
Also addressed this with a new structure.
https://github.com/autowarefoundation/autoware.universe/pull/8300/commits/bafaea1f1154747960e4751678a3c9b4c1f0a903